java/c/src/main/java/org/apache/arrow/c/CDataArrayHandle.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.arrow.c;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.ForeignAllocationManager;
+
+/** Manage the reference count of an imported C Data ArrowArray. */
```

Comment:
Looks good, but I'm a bit concerned that the class name and the fact that it does not implement ReferenceManager obscures the fact that it does reference counting.

Reply:
There is a fundamental mismatch in memory handling designs and this cannot implement ReferenceManager. I will update the docstring and reconsider the name but I do not think anything else is possible, not without entirely overhauling the core implementation. Even if in theory we can work with only the interfaces provided, in practice we need to fit into the concrete implementations, and so BufferLedger is unavoidable.

Reply:
that's fine.

Reply:
I did look at it a little bit. But I don't think you can use any implementation of ReferenceManager but BufferLedger if you want everything else to work. It does feel bad to have effectively two redundant reference counts.

Reply:
I understand. It was a 'it would be nice if....' comment